descriptions.go

```diff
@@ -83,11 +86,11 @@ type CredentialType struct {
 	BackgroundGradientEnd   string
 
 	IsInCredentialStore bool
-	Category            TranslatedString
-	FAQIntro            TranslatedString
-	FAQPurpose          TranslatedString
-	FAQContent          TranslatedString
-	FAQHowto            TranslatedString
+	Category            *TranslatedString
+	FAQIntro            *TranslatedString
+	FAQPurpose          *TranslatedString
+	FAQContent          *TranslatedString
+	FAQHowto            *TranslatedString
```

Comment:
After this change we don't have the requirement anymore that a translation is present for every language, right? So why are all these pointers still needed? In the case where there is no translation, you could simply assign the empty map. `omitEmpty` also omits the empty map, so on that point there is no difference.

Reply:
(Terminology: given a credential type, let us denote with "specified languages" the languages specified in the `<Languages>` tag of that credential type, *or* its issuer if the credential type does not have that tag, *or* its scheme if the issuer does not have that tag.)

These are fields for which it is allowed that they are empty regardless of the specified languages. That's what the pointer specifies:
* Fields of type `TranslatedString` (e.g. the `<Name>` of a credential type) must always contain all of the specified languages,
* Fields of type `*TranslatedString` (such as these ones) may have the null pointer as value regardless of the specified languages.

Reply:
Yes that's what I meant indeed. Besides how it's printed in XML or JSON, there is no real difference anymore between a field not being present, and the empty map right. Or is there a slightly different meaning?

Because now you need to check at all places both whether or not the field is `nil` and whether the particular language is available. Only checking the latter could also satisfy if there is no different meaning.

Reply:
The point of this pointer is the following: some fields must always be translated (according to the specified languages), and some fields may always be empty. We want some way to encode that distinction using the type system, and a pointer seems a fine choice to me for that. It's the same as `int`, which always has some value, and `*int`, where `null` indicates absence.

> there is no real difference anymore between a field not being present, and the empty map right

Well, I would say there is a difference: a nil map is not the same as an empty map. The former may be fine (or not, depending on the field), but the latter is strange: if you meant to not specify translations, `nil` would be more explicit. An empty map to me looks like you forgot to fill it. So I would say, you either say `nil`, or you give a non-empty map.

Reply:
Correspondingly, taking e.g. `FAQIntro`, in XML you either say nothing or you specify an `<FAQIntro>` containing at least one language. That's what this code currently enforces.

Reply:
That would make sense, but I don't see any check doing this. When an empty `TranslatedValue` is passed and no languages are required, then this would be accepted now I think. Or do I overlook a check?

Reply:
* In case of `*TranslatedString`, [here](https://github.com/privacybydesign/irmago/blob/improve-scheme-language-validation/irmaconfig.go#L775) we allow it to be nil;
* If it is not nil, then [a few lines below](https://github.com/privacybydesign/irmago/blob/improve-scheme-language-validation/irmaconfig.go#L783) we invoke the function that checks that it contains the required languages. In particular, if the map is empty then this function will complain (because an empty map does not contain the required languages).

Reply:
On the other hand, if no languages are defined, then indeed that would be accepted. But that to me is one of the features of this PR. It will solve all the senseless warnings when you run `irma scheme verify` on `irma-demo`. And if no languages are defined, what languages could we enforce to be present? 🙂

Reply:
So you say here

> An empty map to me looks like you forgot to fill it.

But currently, we allow this situation to happen in the XML. That's a bit strange then right?

Maybe another question, do we enforce `Languages` to be defined for a `SchemeManager`? If that is the case, then we already prevent the case that we don't know which languages we could expect.

Reply:
Ah, now I finally understand what you mean 🙂 Sorry that took a while. We do allow that to happen, yes, but only when no languages are defined (which is by itself allowed). But even then it makes no sense to allow that, that's a good point. I have fixed it.